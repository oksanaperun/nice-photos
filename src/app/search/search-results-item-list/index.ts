export * from './search-results-item-list.component';
