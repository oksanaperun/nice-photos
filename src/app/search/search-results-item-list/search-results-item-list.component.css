mat-grid-list {
  width: 90%;
  left: 5%;
}
