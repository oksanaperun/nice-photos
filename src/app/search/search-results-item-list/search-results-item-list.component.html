<mat-grid-list cols="3" gutterSize="15px" rowHeight="320px">
  <mat-grid-tile *ngFor="let item of items">
    <app-search-results-item [item]="item"></app-search-results-item>
  </mat-grid-tile>
</mat-grid-list>
