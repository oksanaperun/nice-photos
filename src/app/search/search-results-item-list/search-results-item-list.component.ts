import { Component, Input, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-search-results-item-list',
  templateUrl: './search-results-item-list.component.html',
  styleUrls: ['./search-results-item-list.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})

export class SearchResultsItemListComponent {
  @Input() items; // TODO: add type after search component refactoring
}
