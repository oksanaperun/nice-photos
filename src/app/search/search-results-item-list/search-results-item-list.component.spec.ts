import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MatGridListModule } from '@angular/material';
import { SearchResultsItemListComponent } from './search-results-item-list.component';

describe('SearchResultsItemListComponent', () => {
  const photoWithValidImage = { smallUrl: 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==' };
  let component: SearchResultsItemListComponent;
  let fixture: ComponentFixture<SearchResultsItemListComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchResultsItemListComponent],
      imports: [MatGridListModule],
      schemas: [NO_ERRORS_SCHEMA]
    });

    fixture = TestBed.createComponent(SearchResultsItemListComponent);
    component = fixture.componentInstance;
  });

  it('should render photo tiles', () => {
    component.items = [photoWithValidImage, photoWithValidImage, photoWithValidImage];
    fixture.detectChanges();

    const debugElems = fixture.debugElement.queryAll(By.css('mat-grid-tile'));

    expect(debugElems.length).toBe(component.items.length);
  });
});
