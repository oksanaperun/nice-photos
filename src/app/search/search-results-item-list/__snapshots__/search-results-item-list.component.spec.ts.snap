// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsItemListComponent should render items list 1`] = `
<app-search-results-item-list
  items={[Function Array]}
>
  <mat-grid-list
    cols="3"
    guttersize="15px"
    rowheight="320px"
  >
    
    
    
    <mat-grid-tile>
      
      
      <app-search-results-item />
      
    
    </mat-grid-tile>
    <mat-grid-tile>
      
      
      <app-search-results-item />
      
    
    </mat-grid-tile>
    <mat-grid-tile>
      
      
      <app-search-results-item />
      
    
    </mat-grid-tile>
    
  
  </mat-grid-list>
  
</app-search-results-item-list>
`;
